### ISSUE TITLE
bug: incorrect component type paths w/ dist-custom-elements

### ISSUE THREAD

**Issue by benelan** [ionic-team/stencil#3464]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.17.1

### Current Behavior

The compiler is using the component's tag name instead of the directory and filename when generating the type declaration file  w/ dist-custom-elements.

For example, if your component tag is `my-component`, but you drop the `my-` prefix in the files/directories, the path will still be generated as

```js
// dist/components/index.d.ts
export { MyComponent as MyComponent } from '../types/components/my-component/my-component';
```

### Expected Behavior

In the above example, the path should be generated as

```js
// dist/components/index.d.ts
export { MyComponent as MyComponent } from '../types/components/component/component';
```

### Steps to Reproduce

```bash
git clone https://github.com/benelan/stencil-component-starter.git
cd stencil-component-starter
npm install
npm run build
head -2 dist/components/index.d.ts
[ ! -d "src/components/my-component" ] && echo "Directory src/components/my-component DOES NOT exists."
[ -d "src/components/component" ] && echo "Directory src/components/component exists."
```

### Code Reproduction URL

https://github.com/benelan/stencil-component-starter

### Additional Information

This first occured in `@stencil/core@2.17.0`, likely related to this line: 
https://github.com/ionic-team/stencil/commit/9d3bf154f99abbb6228df5456f318a49f8333362#diff-bd895977bcfc78d28fcf6905fc98b29c4e418008dcdb24bf177594e716b90bcfR59

**Comment by alicewriteswrongs** [comment]
thanks for filing this issue! I've confirmed that this is happening and we'll be working on a fix shortly.

**Comment by benelan** [comment]
Thanks @alicewriteswrongs! Even without the typedef issue, our users may need to use something besides `dist/components/index` to get `setAssetsPath`, since otherwise it'd include all the components in the bundled output making it a lot bigger. 

Would your team consider making the changes in https://github.com/ionic-team/stencil/commit/ff0e8cc54e5e68631cd83302d59b19f2626d43cb opt in or out? Maybe with a config flag. I can create a separate issue for that if it's a possibility.

**Comment by alicewriteswrongs** [comment]
@benelan a quick update — a [pre-release of stencil](https://github.com/ionic-team/stencil/releases/tag/v2.17.2-0) was [published to npm](https://www.npmjs.com/package/@stencil/core/v/2.17.2-0) today which includes the fix that we merged last week (#3468) for this issue. If you have the time to try it out do let us know if it fixes the issue for you!

**Comment by benelan** [comment]
Yup the issue is resolved in `v2.17.2-0`, thank you!

**Comment by rwaskiewicz** [comment]
The fix for this issue has been officially released as a part of [Stencil v2.17.2](https://github.com/ionic-team/stencil/releases/tag/v2.17.2). As a result, I'm going to close this issue and mark it as completed. If the bug re-appears, please feel free to open a new issue!
